src/VaultSharp/V1/SecretsEngines/PKI/GenerateRootRequest.cs

```diff
@@ -0,0 +1,218 @@
+
+        /// <summary>
+        /// Domains for which this certificate is allowed to sign or issue child certificates. 
+        /// If set, all DNS names (subject and alt) on child certs must be exact matches or subsets of the given domains 
+        /// (see https://tools.ietf.org/html/rfc5280#section-4.2.1.10).
+        /// </summary>
+        [JsonPropertyName("permitted_dns_domains")]
+        public List<string> PermittedDnsDomains { get; set; }
+
+        /// <summary>
+        /// If set, Postal Code will be set to this value.
+        /// </summary>
+        [JsonPropertyName("postal_code")]
+        public List<string> PostalCode { get; set; }
+
+        /// <summary>
+        /// Format for the returned private key. 
+        /// Generally the default will be controlled by the "format" parameter as either base64-encoded DER or PEM-encoded DER. 
+        /// However, this can be set to "pkcs8" to have the returned private key contain base64-encoded pkcs8 or PEM-encoded pkcs8 instead. 
+        /// Defaults to "der".
+        /// </summary>
+        [JsonPropertyName("private_key_format")]
+        public string PrivateKeyFormat { get; set; }
+
+        /// <summary>
+        /// If set, Province will be set to this value.
+        /// </summary>
+        [JsonPropertyName("province")]
+        public List<string> Province { get; set; }
+
+        /// <summary>
+        /// The Subject's requested serial number, if any. See RFC 4519 Section 2.31 'serialNumber' for a description of this field. 
+        /// If you want more than one, specify alternative names in the alt_names map using OID 2.5.4.5. 
+        /// This has no impact on the final certificate's Serial Number field.
+        /// </summary>
+        [JsonPropertyName("serial_number")]
+        public string SerialNumber { get; set; }
+
+        /// <summary>
+        /// The number of bits to use in the signature algorithm; accepts 256 for SHA-2-256, 384 for SHA-2-384, and 512 for SHA-2-512. 
+        /// Defaults to 0 to automatically detect based on key length (SHA-2-256 for RSA keys, and matching the curve size for NIST P-Curves).
+        /// </summary>
+        [JsonPropertyName("signature_bits")]
+        public int SignatureBits { get; set; }
+
+        /// <summary>
+        /// If set, Street Address will be set to this value.
+        /// </summary>
+        [JsonPropertyName("street_address")]
+        public List<string> StreetAddress { get; set; }
+
+        /// <summary>
+        /// The requested Time To Live for the certificate; sets the expiration date. 
+        /// If not specified the role default, backend default, or system default TTL is used, in that order. Cannot be larger than the mount max TTL. 
+        /// Note: this only has an effect when generating a CA cert or signing a CA cert, not when generating a CSR for an intermediate CA.
+        /// </summary>
+        [JsonPropertyName("ttl")]
+        public string TTL { get; set; }
```

Comment:
Following the convention used elsewhere in the project. Can we rename this to `TimeToLive`